### ISSUE TITLE
LRP plugin: Quality and output-token models - Add the three requested model backends

### ISSUE THREAD

**Issue by chetan-metrum-ai** [metrum-ai/router#137]
Part of #129. Priority: P2. Related prior work: #15, #24, and #33. Extend that work; do not recreate it.

## Current behavior

Training fits LightGBM quality models at [lrp/train.py:175](https://github.com/metrum-ai/router/blob/5ba920bf3bc6db4463b84f53dac1b5b2e6aa1358/services/learned-routing-policy/lrp/train.py#L175) and log-output-token regression at [lrp/train.py:424](https://github.com/metrum-ai/router/blob/5ba920bf3bc6db4463b84f53dac1b5b2e6aa1358/services/learned-routing-policy/lrp/train.py#L424). Bundle loading constructs LightGBM boosters at [lrp/bundle.py:489](https://github.com/metrum-ai/router/blob/5ba920bf3bc6db4463b84f53dac1b5b2e6aa1358/services/learned-routing-policy/lrp/bundle.py#L489).

## Motivation

An operator wants to use an existing approved tabular model or a simple lookup table for a small workload.

## Proposed protocol

Proposed Python shape. Referenced data types are design placeholders, not existing API claims.
```python
from typing import Protocol
class PredictionModel(Protocol):
    def fit(self, features: Matrix, labels: Vector, weights: Vector) -> ModelArtifact: ...
    def predict(self, features: Matrix) -> Vector: ...
```
Add sklearn, XGBoost, and static lookup table implementations for quality and output-token models. Keep the current LightGBM path compatible.
Proposed config in `services/learned-routing-policy/lrp.example.yaml`: `lrp.models.quality.backend`; `lrp.models.output_tokens.backend`.
Fingerprint: plugin ID, distribution/version, implementation hash, non-secret config hash; backend/version, artifact hash, model parameters, feature order, target transform, supported devices. Declare supported devices (`cpu`, `cuda`, `rocm`); record train and serve device classes. Bind every plugin fingerprint into the bundle manifest under the existing Ed25519 signing path.

## Safety and boundary notes

Receive approved feature matrices and label/weight arrays only. Artifacts are executable trust material; validate provenance and signed hashes before loading. No request payloads or secrets. Plugins never widen the safe-context contract or receive secrets. No changes to `external_policy_context.go` are proposed.

## Acceptance criteria

- [ ] Test fit/load/predict parity for exactly the named alternatives and both output roles.
- [ ] Validate finite quality in [0,1], nonnegative token predictions, and feature/target identity mismatches.
- [ ] Preserve existing training-readiness and missing-label rules.
- [ ] Test manifest fingerprint validation and signed-bundle tampering; document the protocol and sample config.

## Out of scope

Additional backends or model-family variants.

## Open questions

Which single sklearn estimator is the minimal supported implementation? Define the safe serialization contract and static-table key/miss behavior before implementation.
